src/oldreader_api.cpp

```diff
@@ -66,24 +66,26 @@ std::string oldreader_api::retrieve_auth() {
 	}
 
 	std::string pass = cfg->get_configvalue("oldreader-password");
-    std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
-    std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
+	std::string pass_file = cfg->get_configvalue("oldreader-passwordfile");
+	std::string pass_eval = cfg->get_configvalue("oldreader-passwordeval");
 	if ( pass == "" ) {
-        if ( pass_file != "" ) {
-            pass = read_password(pass_file.c_str());
-        } else if ( pass_eval != "") {
-            pass = eval_password(pass_eval.c_str());
-        }
+		if ( pass_file != "" ) {
+			pass = read_password(pass_file);
+			if ( pass == "" ) return "";
+		} else if ( pass_eval != "") {
+			pass = eval_password(pass_eval);
+			if ( pass == "" ) return "";
```

Comment:
Sorry, I missed your edit because GitHub doesn't notify about them. Good thing I found it strange that you abruptly fell silent :)

You're right; it was me who was misunderstanding the code. Sorry about that!

So the code as it currently stands is fine. It'll return if any of `passwordfile` and `passwordeval` are set but return empty string, which is fine; and it'll fall back onto reading stdin if `password` is empty and neither `passwordfile` nor `passwordeval` are set.